app/src/models/branch.ts

```diff
@@ -68,6 +70,20 @@ export class Branch {
     return pieces[1]
   }
 
+  /** The name of remote for a remote branch. If local, will return null. */
+  public get remoteName(): string | null {
+    if (this.type === BranchType.Local) {
+      return null
+    }
+    const remoteRefPrefix = 'refs/remotes/'
+    const remoteBranchName = this.ref.replace(remoteRefPrefix, '')
+    const pieces = remoteBranchName.match(/(.*?)\/.*/)
```

Comment:
I think this could be safer:
```suggestion
    const pieces = this.ref.match(/^refs\/remotes\/(.*?)\/.*/)
```
It makes sure the ref _actually has_ `refs/remotes` as a prefix (in your `replace`, nothing guaranteed the remote prefix was present nor that it was at the beginning).

Reply:
Makes sense